src/dbnode/storage/limits/types.go

```diff
@@ -52,14 +52,18 @@ type QueryLimits interface {
 
 // LookbackLimit provides an interface for a specific query limit.
 type LookbackLimit interface {
+	// Options returns the current limit options.
+	Options() LookbackLimitOptions
 	// Inc increments the recent value for the limit.
 	Inc(new int, source []byte) error
+	// Update changes the lookback limit settings.
+	Update(opts LookbackLimitOptions) error
 }
 
 // LookbackLimitOptions holds options for a lookback limit to be enforced.
 type LookbackLimitOptions struct {
 	// Limit past which errors will be returned.
-	Limit int64
+	Limit *int64
```

Comment:
why did this have to change? so that can actually support a zero value limit? is this going to be backwards compatible with existing limits of 0 meaning off?

Reply:
Hm yeah good point. That is a bummer because `nil` being off is more intuitive. But I'll make the change safe.

Reply:
Added `ForceExceeded` to retain existing zero (disabled) behavior